honeycombio/honeycomb-opentelemetry-web
# feat: ✨ Add `skipOptionsValidation` option and validate options

## Which problem is this PR solving?

- Closes #68 

Adds validation warnings for options. 
Adds `skipOptionsValidation` option to config. This option is used when sending directly to a collector so warnings for not providing an `apiKey` can be suppressed.

## Short description of the changes
- Warns if there is an `apiKey` missing
- Warns if there is a `serviceName` missing
- Warns if a `dataset` is provided but the `apiKey` is a E&S key and a `serviceName is provided
- Warns if `dataset` is missing if a Classic `apiKey` is provided
- Warns if `sampler` has been overridden
- Adds `skipOptionsValidation` option that doesn't show any of these warnings  if set to `true`, set to `false` by default

## How to verify that this has the expected result
- Run the example app and don't provide an `apiKey`, you should see the `apiKey` warning and so forth
- Run the example app and set `skipOptionsValidation` to `true`, you should see the skip options validation message and no other warnings
- Tests pass
